tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -18,131 +18,124 @@
+        public static LiveDebugger Instance { get; private set; }
 
-        private static LiveDebugger Create()
-        {
-            var source = GlobalSettings.CreateDefaultConfigurationSource();
-            var settings = ImmutableDebuggerSettings.Create(DebuggerSettings.FromSource(source));
-            if (!settings.Enabled)
-            {
-                return Create(settings, null, null, null, null, null);
-            }
-
-            var apiFactory = DebuggerTransportStrategy.Get(settings.AgentUri);
-            IDiscoveryService discoveryService = DiscoveryService.Create(source, apiFactory);
-
-            var probeConfigurationApi = ProbeConfigurationApiFactory.Create(settings, apiFactory, discoveryService);
-            var updater = ConfigurationUpdater.Create(settings);
-            IConfigurationPoller configurationPoller = ConfigurationPoller.Create(probeConfigurationApi, updater, settings);
-
-            var snapshotStatusSink = SnapshotSink.Create(settings);
-            var probeStatusSink = ProbeStatusSink.Create(settings);
-
-            var batchApi = BatchUploadApiFactory.Create(settings, apiFactory, discoveryService);
-            var batchUploader = BatchUploader.Create(batchApi);
-            var debuggerSink = DebuggerSink.Create(snapshotStatusSink, probeStatusSink, settings, batchUploader);
+        public static Product Product { get; } = new("LIVE_DEBUGGING");
 
-            var lineProbeResolver = LineProbeResolver.Create();
-            var probeStatusPoller = ProbeStatusPoller.Create(settings, probeStatusSink);
-
-            return Create(settings, discoveryService, configurationPoller, lineProbeResolver, debuggerSink, probeStatusPoller);
-        }
-
-        // for tests
-        internal static LiveDebugger Create(ImmutableDebuggerSettings settings, IDiscoveryService discoveryService, IConfigurationPoller configurationPoller, ILineProbeResolver lineProbeResolver, IDebuggerSink debuggerSink, IProbeStatusPoller probeStatusPoller)
+        public static LiveDebugger Create(
+            ImmutableDebuggerSettings settings,
+            IDiscoveryService discoveryService,
+            ILineProbeResolver lineProbeResolver,
+            IDebuggerSink debuggerSink,
+            IProbeStatusPoller probeStatusPoller)
         {
-            var debugger = new LiveDebugger(settings, discoveryService, configurationPoller, lineProbeResolver, debuggerSink, probeStatusPoller);
-            debugger.Initialize();
+            lock (Lock)
+            {
+                if (Instance is null)
+                {
+                    var updater = ConfigurationUpdater.Create(settings);
+                    Instance = new LiveDebugger(settings, discoveryService, lineProbeResolver, debuggerSink, probeStatusPoller, updater);
+                }
 
-            return debugger;
+                return Instance;
+            }
         }
 
-        private void Initialize()
+        public async Task InitializeAsync()
```

Comment:
How do you eliminate/deal with calling `InitializeAsync` more than once? If it's publicly exposed then there is nothing preventing misuses.